src/check_missing_alleles.py

```diff
@@ -0,0 +1,439 @@
+            mhc_class = "non-classical MHC"
+
+        label = f"HLA-{locus} locus"
+        synonyms = ""
+        class_type = "subclass"
+        parent = f"human {mhc_class} locus"
+        new_loci_rows.add((label, synonyms, class_type, parent))
+
+    with open("../ontology/genetic-locus.tsv", "a+") as fh:
+        for row in new_loci_rows:
+            fh.write(("\t").join(row) + "\n")
+
+    return new_loci
+
+
+def update_index(missing_alleles, missing_genes, missing_loci, missing_molecules):
+    """Adds new entries from IMGT to index.tsv to allow ROBOT to build owl file
+    """
+    mro_labels = set()
+    with open("../index.tsv") as fh:
+        for _ in range(2):
+            next(fh)
+        for line in fh:
+            curr_mro_id = line.rstrip().split("\t")[0]
+            curr_mro_label = line.rstrip().split("\t")[1]
+            mro_labels.add(curr_mro_label)
+            curr_mro_num = int(curr_mro_id.split(":")[1])
+
+    # Next few blocks will iterate MRO ID and pad left with 0s to 7 digits
+    new_tups = []
+    for allele in missing_alleles:
+        chain_name = f"HLA-{allele} chain"
+        if chain_name not in mro_labels:
+            mro_id = f"MRO:{str(curr_mro_num + 1).zfill(7)}"
+            curr_mro_num += 1
+            new_tups.append((mro_id, chain_name, "owl:Class"))
+
+    for gene in missing_genes:
+        gene_name = f"HLA-{gene} chain"
+        if gene_name not in mro_labels:
+            mro_id = f"MRO:{str(curr_mro_num + 1).zfill(7)}"
+            curr_mro_num += 1
+            new_tups.append((mro_id, gene_name, "owl:Class"))
+
+    for locus in missing_loci:
+        locus_name = f"HLA-{locus} locus"
+        if locus_name not in mro_labels:
+            mro_id = f"MRO:{str(curr_mro_num + 1).zfill(7)}"
+            curr_mro_num += 1
+            new_tups.append((mro_id, locus_name, "owl:Class"))
+
+    for molecule in missing_molecules:
+        molecule_name = f"{molecule}"
+        if molecule_name not in mro_labels:
+            mro_id = f"MRO:{str(curr_mro_num + 1).zfill(7)}"
+            curr_mro_num += 1
+            new_tups.append((mro_id, molecule_name, "owl:Class"))
+
+    with open("../index.tsv", "a+") as fh:
+        for tup in new_tups:
+            fh.write(("\t").join(tup) + "\n")
+
+
+def create_classI_prot(missing_chainI):
+    """Creates entries for MHC class I molecules
+
+    Returns:
+        A list of new entries for molecule.tsv
+    """
+    new_classI_molecules = set()
+    with open("../ontology/molecule.tsv", "a+") as fh:
+        for allele in missing_chainI:
+            label = f"HLA-{allele} protein complex"
+            iedb_name = f"HLA-{allele}"
+            synonym = ""
+            restrict_lvl = "complete molecule"
+            class_type = "equivalent"
+            parent = "MHC class I protein complex"
+            taxon = "human"
+            alpha_chain = f"HLA-{allele} chain"
+            beta_chain = "Beta-2-microglobulin"
+            fh.write(
```

Comment:
`csv.writer(fh, delimiter="\t", lineterminator="\n")` might be a better choice here